models/fractal/polygon/fractal_polygon_events.sql

```diff
@@ -0,0 +1,122 @@
+{{ config(
+    schema = 'fractal_polygon',
+    alias = 'events',
+    partition_by = ['block_date'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_time', 'unique_trade_id'],
+    post_hook='{{ expose_spells(\'["polygon"]\'
+                              "project",
+                              "fractal",
+                              \'["springzh"]\') }}'
+    )
+}}
+
+{% set nft_start_date = "2022-12-30" %}
+
+WITH listing_detail AS (
+    SELECT assetContract AS nft_contract_address,
+        contract_address,
+        evt_block_number,
+        evt_block_time,
+        evt_index,
+        evt_tx_hash,
+        lister,
+        listingId,
+        listing:tokenOwner AS tokenOwner,
+        listing:tokenId AS tokenId,
+        listing:startTime AS startTime,
+        listing:endTime AS endTime,
+        listing:quantity AS quantity,
+        listing:currency AS currency,
+        listing:reservePricePerToken AS reservePricePerToken,
+        listing:buyoutPricePerToken AS buyoutPricePerToken,
+        listing:tokenType AS tokenType,
+        listing:listingType AS listingType
+    FROM {{ source ('fractal_polygon', 'Marketplace_evt_ListingAdded') }}
+),
+
+trades AS (
+    SELECT 'buy' AS trade_category,
+        s.evt_block_time,
+        s.evt_block_number,
+        s.evt_tx_hash,
+        s.contract_address,
+        s.evt_index,
+        'Trade' AS evt_type,
+        s.buyer,
+        s.lister AS seller,
+        s.assetContract AS nft_contract_address,
+        l.tokenId AS token_id,
+        s.quantityBought AS number_of_items,
+        'erc721' AS token_standard, -- All of erc721 type
+        '0x0d500b1d8e8ef31e21c99d1db9a6444d3adf1270' AS currency_contract, -- All sale are in MATIC
+        s.totalPricePaid AS amount_raw
+    FROM {{ source ('fractal_polygon', 'Marketplace_evt_NewSale') }} s
+    INNER JOIN listing_detail l ON s.listingId = l.listingId
+    WHERE 1 = 1
+        {% if not is_incremental() %}
+        AND s.evt_block_time >= '{{nft_start_date}}'
+        {% endif %}
+        {% if is_incremental() %}
+        AND s.evt_block_time >= date_trunc("day", now() - interval '1 week')
+        {% endif %}
+)
+
+SELECT
+    'polygon' AS blockchain,
+    'fractal' AS project,
+    'v1' AS version,
+    a.evt_tx_hash AS tx_hash,
+    date_trunc('day', a.evt_block_time) AS block_date,
+    a.evt_block_time AS block_time,
+    a.evt_block_number AS block_number,
+    amount_raw / power(10, p.decimals) * p.price AS amount_usd,
+    amount_raw / power(10, p.decimals) AS amount_original,
+    amount_raw,
+    p.symbol AS currency_symbol,
```

Comment:
we should use tokens.erc20 instead of prices as a source for decimals and token symbol

Reply:
Updated.